### DISCUSSION TITLE
Auth confusion, Next.js 14 app router

### DISCUSSION THREAD

**Discussion by yuriipalam** [pocketbase/pocketbase#5602] [Q&A] [answered]
I'm using PocketBase for authentication in my Next.js 14 app, but I'm a bit confused about handling token validation and refresh in the middleware. My code:
```ts
const tokenCookie = request.cookies.get(PocketBaseAuthCookieKey);
const pb = await getServerPocketBase(tokenCookie);
try {
    pb.authStore.loadFromCookie(`${tokenCookie.name}=${tokenCookie.value}`);
    pb.authStore.isValid && (await pb.collection("users").authRefresh());
    // go ahead
  } catch (_) {
    pb.authStore.clear();
    // redirect if route protected
}
```

The problem is that this code refreshes the token on every single request, because `authStore.isValid` only checks the token payload without actually validating it. I understand that usually, a JWT access token is refreshed only when expired, but not when valid. `authRefresh()` method seems to take 300-500ms in my case, which impacts performance significantly.

- Is there a more efficient way to handle token validation and refresh in PocketBase, especially in a server-side context like Next.js?
- How can I avoid calling `authRefresh()` on every request?

Or should I simply avoid using PocketBase in server-side context? Perhaps it makes everything so easier...

**Comment by thangngoc89** [comment, marked as answer]
I would recommend you to give this a read. Overall, SSR + Pocketbase is not a good idea.

https://github.com/pocketbase/pocketbase/discussions/5313

**Reply by yuriipalam** [reply]
Thanks for your answer, very useful read. I should probably rethink my app's architecture.
